iati/core/tests/test_rulesets.py

```diff
@@ -220,32 +240,44 @@ def test_rule_paths(self, basic_rule):
                 assert path.startswith(basic_rule.xpath_base)
 
     def test_rule_invalid_case(self, rule_constructor, invalid_case):
-        """Check that a rule cannot be instantiated when the case is invalid.
-
-        Todo:
-            Add tests with empty path AND empty xpath_base.
-        """
+        """Check that a rule cannot be instantiated when the case is invalid."""
         xpath_base = 'an xpath'
 
         with pytest.raises(ValueError):
             rule_constructor(xpath_base, invalid_case)
 
     def test_is_valid_for(self, valid_dataset, rule):
-        """Check that a given rule returns the expected result when given a dataset.
-
-        Todo:
-            Maybe too much of a shortcut as can't fully pass until all implementations complete. Possibly the wrong abstraction in the long-term.
-        """
+        """Check that a given rule returns the expected result when given a dataset."""
         assert rule.is_valid_for(valid_dataset)
 
     def test_is_invalid_for(self, invalid_dataset, rule):
-        """Check that a given rule returns the expected result when given a dataset.
-
-        Todo:
-            Maybe too much of a shortcut as can't fully pass until all implementations complete. Possibly the wrong abstraction in the long-term.
-        """
+        """Check that a given rule returns the expected result when given a dataset."""
         assert not rule.is_valid_for(invalid_dataset)
 
+    @pytest.mark.parametrize("junk_data", iati.core.tests.utilities.find_parameter_by_type([], False))
+    def test_is_valid_for_raises_error_on_non_permitted_argument(self, rule, junk_data):
+        """Check that a given rule returns expected error when passed an argument that is not a dataset."""
+        with pytest.raises(AttributeError):
+            rule.is_valid_for(junk_data)
+
+    def test_is_valid_for_raises_error_when_passed_an_etree(self, rule):
+        """Check that an error is raised if an etree is given as an argument instead of a dataset."""
+        with pytest.raises(AttributeError):
+            rule.is_valid_for(iati.core.resources.load_as_tree(iati.core.resources.get_test_data_path('valid_atleastone')))  # use more generic dataset
+
+    def test_empty_xpath_base_rule_init_normalised_paths(self, rule_empty_xpath_base):
+        """Check that a rule with an empty xpath_base can be instantiated correctly with normalised `paths`."""
+        # Exclude rules without `paths`.
+        if 'paths' in dir(rule_empty_xpath_base):
+            for path in rule_empty_xpath_base.paths:
+                assert path.startswith(rule_empty_xpath_base.xpath_base)
+
+    def test_empty_xpath_base_empty_path_is_valid_for(self, rule_empty_xpath_base_and_paths, valid_dataset):
+        """Check that a rule with an empty xpath_base and empty paths raises error."""
+        # RuleStartswith raises an IndexError instead.
+        with pytest.raises((lxml.etree.XPathEvalError, IndexError)):
```

Comment:
The API for the different Rule subclasses should be consistent.

This relates somewhat to the TODO about maybe hiding `lxml.etree.XPathEvalError`, but would suggest making that the desired exception (so either `try-except`ing ths `IndexError` and raising the other one; or (probably better) making a lxml call that raises the exception itself)